src/vs/server/node/webClientServer.ts

```diff
@@ -128,6 +139,38 @@ export class WebClientServer {
 		return serveFile(this._logService, req, res, filePath, headers);
 	}
 
+	/**
+	 * Handle extension resources
+	 */
+	private async _handleWebExtensionResource(req: http.IncomingMessage, res: http.ServerResponse, parsedUrl: url.UrlWithParsedQuery): Promise<void> {
+		// Strip `/web-extension-resource/` from the path
+		const normalizedPathname = decodeURIComponent(parsedUrl.pathname!); // support paths that are uri-encoded (e.g. spaces => %20)
```

Comment:
This is so involved to get back to the original URL, can this be encoded in a query parameter rather? Or is the template a strict prefix that paths are concatenated to.

Reply:
> Or is the template a strict prefix that paths are concatenated to.

Yes. Hence it is made part of the `path` like `static` urls.